encoder/tests/conflictEnumDeferror.txtar

```diff
@@ -0,0 +1,19 @@
+
+This would be the typical cases we should fail, since Bar is refereced to a type with default
+and we are trying to assign another default
+
+Seems CUE is picking the default before eval it, it seems we can get the reference first and
+check whether default exist there, if yes if the default also exist for the current cue object error out
+
+Or should we just forbidden the disjunction between Enum and other types??
```

Comment:
This is a really interesting question! It should probably be a comment in the PR, or even its own issue, rather than hiding it in an actual committed file, i think.

My intuition is that, first and foremost, we have to follow the constraints imposed by TypeScript's own rules. From there, we should do the least magical thing possible. Given that, my initial thought is to allow this, as:

```ts
export enum AOrEnum: {
  Foo: 'foo',
  Bar: 'bar',
  Baz: 'baz'
}

export interface Foo {
  Bar: AOrEnum | "ohai"
}

export const defaultFoo: Foo = {
  Bar: "ohai"
}
```

At first, this seemed reasonable, because of what i felt i could infer about authorial intent:

1. There's an author of `AOrEnum`, who expressed the intent that the default value of their type be `baz`
2. There's an author of `Foo`, who is (clearly) directly aware of `AOrEnum`, as they wrote some text explicitly referencing it.
3. `Foo`'s author, being ostensibly aware of `AOrEnum`, is choosing to supersede the default of `baz` with `ohai`.

However, going through this exercise led me to realize that this case is narrow and circumstantial, not based on generalizable principles of the sort people can reason about. That's because we're bumping into fundamental differences in the models on which CUE and TS are based:

1. Defaults are a native concept in CUE that authors must deal with; no such concept exists in TS
2. The primitive kinds in the languages are simply different (disjunction semantics in CUE; existence of enums in TS)
3. While both TS and CUE are mostly structurally typed, which generally it possible to flow types through the system more freely than if names must strictly align, we're (necessarily) making that harder by trying to snapshot bits of the system to translate from CUE to TS, without ever being able to guarantee that we have the whole type graph to examine for completeness or correctness

i don't know what our good answers are here, yet, but the above sort of reasoning makes me reasonably sure there simply aren't any, and we shouldn't waste time or effort searching for them. Instead, we should be very conservative in what we support, erroring aggressively when the slightest bit of ambiguity arises about how to e.g. handle a default or flow an enum into an interface, and then consider opening up further when use cases present themselves.

That's still not a clear, categorical rule we can apply easily, but...well, it's a general basis for saying no and constraining scope, which is usually a safe place to be.

Reply:
All that said, it's worth noting that CUE seems to follow the same logic about user intent/awareness that i originally ran be - see [this playground](https://tip.cuelang.org/play/?id=0v7nuHFdtS6#cue@export@cue), and note that `C` has a single default, but `D` does not. Which is...actually really interesting, and potentially useful.